java-checks/src/main/java/org/sonar/java/checks/AtLeastOneConstructorCheck.java

```diff
@@ -77,22 +95,17 @@ private void checkClassTree(ClassTree tree) {
 
   private static boolean requiresInitialization(VariableTree variable) {
     Symbol symbol = variable.symbol();
-    return variable.initializer() == null && symbol.isPrivate() && !symbol.isStatic() && isNotEJBAnnotated(symbol) &&!isAutowired(symbol);
+    return variable.initializer() == null && symbol.isPrivate() && !symbol.isStatic() && !isAutowired(symbol);
   }
 
   private static boolean isAutowired(Symbol symbol) {
-    return symbol.metadata().isAnnotatedWith("org.springframework.beans.factory.annotation.Autowired") ||
-      symbol.metadata().isAnnotatedWith("javax.inject.Inject");
+    return AUTOWIRED_ANNOTATIONS.stream().anyMatch(symbol.metadata()::isAnnotatedWith);
   }
 
   private static boolean isAnnotationExcluded(Symbol symbol) {
     return EXCLUDED_ANNOTATIONS.stream().anyMatch(symbol.metadata()::isAnnotatedWith);
   }
 
-  private static boolean isNotEJBAnnotated(Symbol symbol) {
```

Comment:
By removing this, you will only exclude specifically `javax.ejb.EJB` and not every annotation in `javax.ejb`.
This was probably an approximation to avoid having to go trough all possible annotations in `javax.ejb`.
If you can show me that only `javax.ejb.EJB` should be excluded, removing it is fine, otherwise, we should keep it!

Reply:
Under javax.ejb package only `@EJB` annotation has `@Target` set to `FIELD`, so another annotation from this package can't be used for fields.
https://docs.oracle.com/javaee/7/api/javax/ejb/package-frame.html

Method `isNotEJBAnnotated` is used only for check if field is decorated by given annotation.

So when I added list AUTOWIRED_ANNOTATIONS with more annotation which can be used for fields, this method is not nessesary.
We simpllyfi code in this case.

Reply:
Okay, it looks good in this case!